pkg/controller/operators/olm/operator.go

```diff
@@ -158,6 +182,54 @@ func (a *Operator) syncDeployment(obj interface{}) (syncError error) {
 	return nil
 }
 
+func (a *Operator) syncOperatorGroups(obj interface{}) error {
+	op, ok := obj.(*v1alpha2.OperatorGroup)
+	if !ok {
+		log.Debugf("wrong type: %#v\n", obj)
+		return fmt.Errorf("casting OperatorGroup failed")
+	}
+
+	// write namespace matches to status field
+	selector, err := metav1.LabelSelectorAsSelector(&op.Spec.Selector)
+	if err != nil {
+		return err
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err
+	}
+	op.Status.Namespaces = namespaceList.Items
+
+	// make string list made up of comma separated namespaces
+	var b strings.Builder
+	nsCount := len(namespaceList.Items)
+	for i := 0; i < nsCount-1; i++ {
+		fmt.Fprintf(&b, "%v,", namespaceList.Items[i])
```

Comment:
This might require .Name here, but maybe there's a better way to do all this in general?

Reply:
If you don't need to use .Name (which I think we may need to), you can use `strings.Join(namespaceList.Items, ",")` here as an alternative.

Reply:
I think it makes sense for `namespaces` in the status to just be a list of strings. Normally when we do object references we try and include GVK but namespaces are pretty ingrained (e.g. `metadata.namespace` is just a string on every kube object).

If we make that change, then this part just becomes what @njhale suggested